src/main/java/org/truffleruby/core/rope/OnHeapRope.java

```diff
@@ -0,0 +1,20 @@
+package org.truffleruby.core.rope;
+
+import org.jcodings.Encoding;
+
+public abstract class OnHeapRope extends Rope {
```

Comment:
I have to debate names, but how about `ManagedRope`. Native ropes are also on the heap, and managed ropes can be scalar-replaced.